pennylane/_qubit_device.py

```diff
@@ -647,14 +647,17 @@ def state(self):
         raise NotImplementedError
 
     def density_matrix(self, wires):
-        """Returns the reduced density matrix prior to measurement.
+        """Returns the reduced density matrix over the given wires.
 
-        .. note::
+        Args:
+            wires (Wires): wires of the reduced system
 
-            Only state vector simulators support this property. Please see the
-            plugin documentation for more details.
+        Returns:
+            array[complex]: complex array of shape ``(2 ** len(wires), 2 ** len(wires))``
+            representing the reduced density matrix of the state prior to measurement.
         """
-        raise NotImplementedError
+        state = getattr(self, "state", None)
```

Comment:
Would this be preferred to using `access_state`?

Reply:
I think it should work, and it is what the device is doing in `access_state` but let's try that in a separate PR imo

Reply:
The problem here is that `access_state` calls `density_matrix` therefore making impossible to use it. The way to get the state in `access_state` is the following:

`state = getattr(self, "state", None)`